Изменила ДЗ4

NKVol opened the pull request:
Изменила LionStFMS.py:
1. Изменила конструктор
2. Изменила реализацию fsm_realisation
(Все изменения внесла согласно замечаниям)

NKVol committed:
Добавила ДЗ_4

NKVol committed:
Добавила .travis.yml файл

NKVol committed:
Удалила command to install dependencies

NKVol committed:
Update .travis.yml

NKVol committed:
Update LionSateFMS_Test.py

исправила ошибка в test_init

NKVol committed:
Update LionStFMS.py

исправила ошибки в коде

NKVol committed:
Update README.md

Добавила бейдж

NKVol committed:
Update README.md

исправила бейдж

NKVol committed:
Update README.md

Поправила настройки бейдж Travis ci

NKVol committed:
Update LionStFMS.py

Изменила код конструктора
Добавила метод lion_fms_work(self)

NKVol committed:
Update LionSateFMS_Test.py

Разбила тест FMS на более мелкие
Добавила проверку исходных состояний: test_init_fed, test_init_hungry
Добавила проверку работы FMS для всех состояний в цикле

NKVol committed:
Update LionSateFMS_Test.py

Добавила тест ввода не верного значения объекта: test_fms_bad_obj 
Изменила вид исключения в test_init с NameError на ValueError

NKVol committed:
Update LionStFMS.py

Изменила реализация конечного автомата состояния льва fsm_realisation: полностью поменяла логику
Изменила конструктор: добавила список всех возможных состояний в конструктор: lion_list
Изменила objects_list: добавила объектов
Изменила lion_fsm_work: убрала цикл while

NKVol committed:
Update LionStFMS.py

убрала ненужный метод и список

NKVol committed:
Update LionSateFMS_Test.py

Изменила метод test_fms

NKVol committed:
Update LionStFMS.py

Убрала obj_list

NKVol committed:
Update LionStFMS.py

Изменения:
1.Убрала комментарии
2.Изменила список на словарь в конструкторе
3.Изменила логику FMS

NKVol committed:
Update LionSateFMS_Test.py

Изменения:
1.Убрала комментарии
2.Убрала последний не нужный тест test_fms

NKVol committed:
Update LionStFMS.py

Исправила конструктор и fsm_realisation

skyline-gleb commented:
пока закрываю, вроде дальнейшие улучшения обсудили

NKVol committed:
Update LionStFMS.py

Изменила словарь

NKVol committed:
Update LionSateFMS_Test.py

Добавила списки и словарь
Изменила реализацию тестов
Убрала комментарии

NKVol commented:
Изменила LionStFMS.py:
Изменила словарь
Изменила LionSateFMS_Test.py:
Добавила словарь в тесты
Изменила способ ввода данных в тесты

NKVol committed:
Update LionSateFMS_Test.py

Оптимизировала тесты

NKVol committed:
Update LionStFMS.py

Слегка подкорректировала __init__

NKVol committed:
Update LionStFMS.py

заменила списки на словари в словаре state_list
сделала обращения по ключам

NKVol committed:
Update LionSateFMS_Test.py

Добавила обращение по ключам для наглядности

skyline-gleb commented:
ох как сильно ты все усложнила тем самым давним коммитом https://github.com/2gisprojectT/lesson_4_homework/pull/36/commits/6057f1ac2d12dca2c93f5b4e4ff6fa429d0762fe и я бы откатил назад и не разделал на несколько список, это чересчур, нет?
и использование в тестах циклов это плохо. сути не понимаешь. ты тестами проверяешь что код правильно отработал, а не то что у тебя сытый/голодный и т.п. Задача тестов и использующихся в них входных данных проверить каждое ветвление кода, что оно верно работает. И конечно же сделать это за минимум тестов, что в твоем случае достигается подготовкой тестовых входных данных, минимально необходимых

NKVol committed:
Update LionStFMS.py

Убрала списки

NKVol committed:
Update LionStFMS.py

Код стаил поправила

NKVol committed:
Update LionSateFMS_Test.py

Убрала списки
Убрала лишний тест

NKVol commented:
Внесла изменения с учетом замечаний
Убрала списки, циклы, лишний тест

NKVol committed:
Update LionSateFMS_Test.py

Изменила словарь

NKVol committed:
Update LionStFMS.py

Сделала список входным параметром при создании Льва

NKVol commented:
Изменила LionStFMS.py:
Убрала словарь, сделала его входным параметром при создании Льва

skyline-gleb commented on LionStFMS.py:
```diff
@@ -0,0 +1,15 @@
+
+class LionStateFSM:
+    def __init__(self, state, state_list):
+        self.state_list = state_list
```
рановато присвоила, можно и попозже чуток

NKVol replied:
Согласна, переделала

skyline-gleb commented on LionStFMS.py:
```diff
@@ -0,0 +1,15 @@
+
+
```
эти две пустые строки пайчарм просит делать?

NKVol replied:
Да, иначе говорит не по стилю(

skyline-gleb replied:
ага, это ожидал, типо сверху и бла-бла-бла)

skyline-gleb commented on LionSateFMS_Test.py:
```diff
@@ -0,0 +1,42 @@
+                    'охотник': {'action': 'убежать', 'new_state': 'голодный'},
+                    'дерево': {'action': 'спать', 'new_state': 'голодный'}
+                }
+            }
```
слишком избыточно

NKVol committed:
Update LionStFMS.py

Переместила присвоение ниже

NKVol committed:
Update LionSateFMS_Test.py

Убрала избыточность

NKVol committed:
Update LionSateFMS_Test.py

NKVol commented:
Исправила согласно замечаниям
Со всеми замечаниями согласна

skyline-gleb commented on LionSateFMS_Test.py:
```diff
@@ -0,0 +1,34 @@
+            self.assertEqual(lion.state_list['сытый']['антилопа']['action'], lion.action,
+                             "Ошибка! Должен быть " + lion.state_list['сытый']['антилопа']['action'])
+            self.assertEqual(lion.state_list['сытый']['антилопа']['new_state'], lion.state,
+                             "Ошибка! Должен быть " + lion.state_list['сытый']['антилопа']['new_state'])
```
27 и 25 мне кажется лишнее, так как имя теста говорит за себя

skyline-gleb commented on LionSateFMS_Test.py:
```diff
@@ -0,0 +1,34 @@
+
+    def test_init_good_obj(self):
+            lion = LionStateFSM('сытый', lion_state_list)
+            lion.fsm_realisation('антилопа')
```
а вот вместо 'сытый' и 'антилопа' наверно можно через lion_state_list сделать, тогда тест не надо менять если поменяла таблицу.
либо тогда ниже, в ассертах, не использовать конструкции вида `lion.state_list['сытый']['антилопа']['action']`. Вообщем реши как лучше

skyline-gleb commented on LionSateFMS_Test.py:
```diff
@@ -0,0 +1,34 @@
+    def test_init_good_obj(self):
+            lion = LionStateFSM('сытый', lion_state_list)
+            lion.fsm_realisation('антилопа')
+            self.assertEqual(lion.state_list['сытый']['антилопа']['action'], lion.action,
```
`lion.state_list['сытый']['антилопа']['action']` вот так кстати плохо, у тебя эталон то `lion_state_list`

skyline-gleb commented:
плюс у тебя поехали отступы.
и в тесте test_good_init_state есть недостача

NKVol committed:
Update LionSateFMS_Test.py

NKVol committed:
Update LionStFMS.py

skyline-gleb commented:
очень плохо. подогнала код льва под тесты, когда уже код льва был одобрен.
в тестах впринципе все исправила, кроме того что недостачу так и не нашла.

skyline-gleb commented:
да и блин вообще. есть определенный шаблон входной таблицы переходов/состояний, на котором сошлись, но условились не проверять корректность. в итоге ты же переделала льва после того как упростила таблицу и в итоге пихни ему ту самую и он сломается.

NKVol commented:
Я не нашла, как название ключа в качестве параметра использовать, поэтому и все переделала. Есть возможность в Python брать название ключа?

skyline-gleb commented:
``` python
ict = {"key1": "value", "key2": "value"}
key1 = dict.keys()[0]
```

NKVol committed:
Update LionSateFMS_Test.py

Отредактировала согласно замечаниям:
Вернула словарь, сделала обертку для данных, убрала лишние комментарии, добавила проверки

NKVol committed:
Update LionStFMS.py

Вернула назад работу со словарем, как было;
Добавила проверку словаря, что он не пустой. В дальнейшем нужно сделать проверку формата

NKVol commented:
Update LionSateFMS_Test.py
Отредактировала согласно замечаниям:
Вернула словарь, сделала обертку для данных, убрала лишние комментарии, добавила проверки.

Update LionStFMS.py
Вернула назад работу со словарем, как было;
Добавила проверку словаря, что он не пустой. В дальнейшем нужно сделать проверку формата.

NKVol committed:
Update LionSateFMS_Test.py

Исправила ошибку

NKVol commented:
Update LionSateFMS_Test.py
Отредактировала согласно замечаниям:
Вернула словарь, сделала обертку для данных, убрала лишние комментарии, добавила проверки

Update LionStFMS.py
Вернула назад работу со словарем, как было;
Добавила проверку словаря, что он не пустой. В дальнейшем нужно сделать проверку формата

skyline-gleb commented:
:+1:

NKVol committed:
Update LionSateFMS_Test.py

Добавила проверку state_list в test_good_init_state

NKVol commented:
Добавила проверку state_list в test_good_init_state

skyline-gleb commented:
упорная ты, Наташа :) принятно дубль два :+1:

NKVol committed:
Update LionSateFMS_Test.py

Изменила state_list_wrapper

NKVol commented:
> упорная ты, Наташа :) принятно дубль два :+1:

Есть такое) 
Изменила state_list_wrapper. Вот так еще в Python 3 это можно сделать. Вроде аккуратнее  смотрится )

skyline-gleb commented on LionSateFMS_Test.py:
```diff
@@ -35,13 +33,12 @@ def test_empty_data(self):
-        self.assertEqual(state_list_wrapper['action'], lion.action)
-        self.assertEqual(state_list_wrapper['new_state'], lion.state)
+        self.assertEqual(lion_state_list[state_list_wrapper['state']][state_list_wrapper['object']]['action'], lion.action)
+        self.assertEqual(lion_state_list[state_list_wrapper['state']][state_list_wrapper['object']]['new_state'], lion.state)
```
вот тут то как-то не очень красиво вышло, я оч долго вычитывал где какая скобка чтоб понять кто откуда берется. такое вообще всегда тяжело. надо подумать :)

skyline-gleb replied:
имею ввиду я подумаю. постараюсь не забыть подумать..)))
